src/pages/hosts/HostsTable.tsx

```diff
@@ -1,304 +1,254 @@
-    useTableInputFilter<HostsUrlParam>({
-      urlSearchParam: "hostId",
-      sendAnalyticsEvent: sendHostsTableFilterEvent,
-    });
+  const updateRowSelection = (rowState: RowSelectionState) => {
+    const selectedHosts = Object.keys(rowState).map(
+      (key) => table.getRowModel().rowsById[key]?.original
+    );
+    setSelectedHosts(selectedHosts);
+  };
 
-  // STATUSES URL PARAM
-  const [statusesValue, onChangeStatuses] =
-    useTableCheckboxFilter<HostsUrlParam>({
-      urlSearchParam: "statuses",
-      sendAnalyticsEvent: sendHostsTableFilterEvent,
-    });
+  const updateFilters = (filterState: ColumnFiltersState) => {
+    const updatedParams = { page: "0" };
 
-  // DISTRO URL PARAM
-  const [distroIdValue, onChangeDistroId, updateDistroIdUrlParam] =
-    useTableInputFilter<HostsUrlParam>({
-      urlSearchParam: "distroId",
-      sendAnalyticsEvent: sendHostsTableFilterEvent,
+    filterState.forEach(({ id, value }) => {
+      const key = mapIdToFilterParam[id];
+      updatedParams[key] = value;
     });
 
-  // CURRENT TASK ID URL PARAM
-  const [
-    currentTaskIdValue,
-    onChangeCurrentTaskId,
-    updateCurrentTaskIdUrlParam,
-  ] = useTableInputFilter<HostsUrlParam>({
-    urlSearchParam: "currentTaskId",
-    sendAnalyticsEvent: sendHostsTableFilterEvent,
+    setQueryParams(updatedParams);
+    sendEvent({ name: "Filter Hosts", filterBy: Object.keys(filterState) });
+  };
+
+  const tableContainerRef = useRef<HTMLDivElement>(null);
+  const table = useLeafyGreenTable<Host>({
+    columns,
+    containerRef: tableContainerRef,
+    data: hosts ?? [],
+    defaultColumn: {
+      enableColumnFilter: false,
+      enableSorting: false,
+      // Handle bug in sorting order
+      // https://github.com/TanStack/table/issues/4289
+      sortDescFirst: false,
+    },
+    state: {
+      columnFilters: filters,
+      rowSelection,
+      sorting,
+    },
+    hasSelectableRows: true,
+    manualFiltering: true,
+    manualSorting: true,
+    manualPagination: true,
+    onColumnFiltersChange: onChangeHandler<ColumnFiltersState>(
+      setFilters,
+      (updatedState) => {
+        updateFilters(updatedState);
+        table.resetRowSelection();
+      }
+    ),
+    onRowSelectionChange: onChangeHandler<RowSelectionState>(
+      setRowSelection,
+      updateRowSelection
+    ),
+    onSortingChange: onChangeHandler<SortingState>(
+      setSorting,
+      (updatedState) => {
+        tableSortHandler(updatedState);
+        table.resetRowSelection();
+      }
+    ),
   });
 
-  // OWNER URL PARAM
-  const [ownerValue, onChangeOwner, updateOwnerUrlParam] =
-    useTableInputFilter<HostsUrlParam>({
-      urlSearchParam: "startedBy",
-      sendAnalyticsEvent: sendHostsTableFilterEvent,
-    });
+  return (
+    <BaseTable
+      data-cy="hosts-table"
+      data-loading={loading}
+      loading={loading}
+      shouldAlternateRowColor
+      table={table}
```

Comment:
nit. Can we reflect the expected page size when loading? 
You can use the `loadingRows` prop.